ext/nnstreamer/tensor_filter/tensor_filter_executorch_llama.cc

```diff
@@ -0,0 +1,250 @@
+ * @bug     No known bugs.
+ * @note    Currently experimental.
+ * @details A pipeline example:
+ *          gst-launch-1.0  filesrc location=input.txt ! application/octet-stream !
+ *          tensor_converter ! other/tensors,format=flexible !
+ *          tensor_filter framework=executorch-llama model=model.pte,tokenizer.model invoke-dynamic=true !
+ *          other/tensors,format=flexible ! tensor_decoder mode=octet_stream !
+ *          application/octet-stream ! filesink location=result.txt
+ */
+
+#include <glib.h>
+#include <nnstreamer_cppplugin_api_filter.hh>
+#include <nnstreamer_log.h>
+#include <nnstreamer_plugin_api_util.h>
+#include <nnstreamer_util.h>
+#include <vector>
+
+#include <executorch/extension/module/module.h>
+#include <executorch/extension/tensor/tensor.h>
+
+#include <executorch/examples/models/llama/runner/runner.h>
+
+using namespace torch::executor;
+
+namespace nnstreamer
+{
+namespace tensorfilter_executorch_llama
+{
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+void init_filter_executorch_llama (void) __attribute__ ((constructor));
+void fini_filter_executorch_llama (void) __attribute__ ((destructor));
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
+
+/**
+ * @brief tensor-filter-subplugin concrete class for ExecuTorch-Llama
+ */
+class executorch_llama_subplugin final : public tensor_filter_subplugin
+{
+  private:
+  static const char *fw_name;
+  static executorch_llama_subplugin *registeredRepresentation;
+
+  std::string model_path; /**< The model *.pte file */
+  std::string tokenizer_path; /**< The tokenizer file */
+
+  /* executorch */
+  std::unique_ptr<example::Runner> runner; /**< Runner for Llama model */
+
+  public:
+  static void init_filter_executorch_llama ();
+  static void fini_filter_executorch_llama ();
+
+  executorch_llama_subplugin (){};
+  ~executorch_llama_subplugin (){};
+
+  tensor_filter_subplugin &getEmptyInstance ();
+  void configure_instance (const GstTensorFilterProperties *prop);
+  void invoke (const GstTensorMemory *input, GstTensorMemory *output);
+  void invoke_dynamic (GstTensorFilterProperties *prop,
+      const GstTensorMemory *input, GstTensorMemory *output) override;
+  void getFrameworkInfo (GstTensorFilterFrameworkInfo &info);
+  int getModelInfo (model_info_ops ops, GstTensorsInfo &in_info, GstTensorsInfo &out_info);
+  int eventHandler (event_ops ops, GstTensorFilterFrameworkEventData &data);
+};
+
+const char *executorch_llama_subplugin::fw_name = "executorch-llama";
+
+/**
+ * @brief Method to get empty object.
+ */
+tensor_filter_subplugin &
+executorch_llama_subplugin::getEmptyInstance ()
+{
+  return *(new executorch_llama_subplugin ());
+}
+
+/**
+ * @brief Method to prepare/configure ExecuTorch instance.
+ */
+void
+executorch_llama_subplugin::configure_instance (const GstTensorFilterProperties *prop)
+{
+  try {
+    /* Load model file (.pte) */
+    if (!g_file_test (prop->model_files[0], G_FILE_TEST_IS_REGULAR)) {
```

Comment:
I'm not sure, but should we check this even if verify_model_path is true?

Reply:
Thanks for pointing it. I changed `verify_model_path` as 0.
I found that it does not care `GST_TF_FW_V1` properly. It should be fixed later.

Reply:
Oh, I tested using llama2c filter and indeed `verify_model_path` is not working at v1 😭